File: src/main/java/org/example/collection/Location.java
package org.example.collection;

import java.util.Objects;

/**
 * Класс координаты
 */
public class Location {
    private Double x; //Поле не может быть null
    private Integer y; //Поле не может быть null
    private int z;
    private String name; //Поле может быть null

    public Location() {
    }

    public Location(Double x, Integer y, int z, String name) {
        this.x = x;
        this.y = y;
        this.z = z;
        this.name = name;
    }

    public Double getX() {
        return x;
    }

    public Integer getY() {
        return y;
    }

    public int getZ() {
        return z;
    }

    public String getName() {
        return name;
    }

    public void setY(Integer y) {
        this.y = y;
    }

    public void setX(Double x) {
        this.x = x;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setZ(int z) {
        this.z = z;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Location location = (Location) o;
        return z == location.z && Objects.equals(x, location.x) && Objects.equals(y, location.y) && Objects.equals(name, location.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(x, y, z, name);
    }

    @Override
    public String toString() {
        return "Location{" +
                "x=" + x +
                ", y=" + y +
                ", z=" + z +
                ", name='" + name + '\'' +
                '}';
    }
}

File: common/src/main/kotlin/collection/Location.kt
package collection

import java.io.Serializable
import java.util.Objects

class Location(val x: Double, val y: Int, val z: Int, val name: String) : Serializable {
    companion object {
        const val serialVersionUID = 6157220399850999705
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val location = other as Location
        return z == location.z && Objects.equals(x, location.x) && y == location.y && name == location.name
    }

    override fun hashCode(): Int {
        return Objects.hash(x, y, z, name)
    }

    override fun toString(): String {
        return "Location{" +
            "x=" + x +
            ", y=" + y +
            ", z=" + z +
            ", name='" + name + '\'' +
            '}'
    }
}
